page-h5p-listing.php

```diff
@@ -3,40 +3,65 @@
 <?php
 if ( \PressbooksBook\Helpers\is_book_public() ) :
 	$data = \PressbooksBook\Helpers\get_h5p_activities();
+	$activities = count( $data['activities'] );
+	$filtered_by_id_exists = isset( $_GET['h5p_id'] ) && $activities === 1;
+	$filtered_does_not_exist = isset( $_GET['h5p_id'] ) && $activities === 0;
 	?>
 	<div id="post-<?php the_ID(); ?>" <?php post_class( 'h5p-listing-page' ); ?>>
-			<h1 class="page-title"><?php echo __( 'H5P activities list', 'pressbooks-book' ); ?></h1>
+		<h1 class="page-title">
+			<?php
+			if ( $filtered_by_id_exists ) {
+				echo $data['activities'][0]['title'];
+			} elseif ( $filtered_does_not_exist ) {
+				echo __( 'Invalid H5P activity', 'pressbooks-book' );
+			} else {
+				echo __( 'H5P activities list', 'pressbooks-book' );
+			}
+			?>
+		</h1>
+		<?php if ( ! isset( $_GET['h5p_id'] ) ) : ?>
 			<p class="h5p-count">
 				<?php echo __( 'This book includes ', 'pressbooks-book' ); ?>
 				<strong><?php echo $data['total'] . ' '; ?></strong>
-			<?php echo __( 'H5P activities. Only those which have been inserted into book content will be included if the book is cloned.', 'pressbooks-book' ); ?>
-			<?php echo sprintf( '<button type="button" id="h5p-show-hide" class="btn btn-secondary btn-sm" show-all-text="%s" hide-all-text="%s" aria-label="%s"></button>', __( 'Expand all', 'pressbooks-book' ), __( 'Hide all', 'pressbooks-book' ), __( 'Expand all activities', 'pressbooks-book' ) ); ?>
+				<?php echo __( 'H5P activities. Only those which have been inserted into book content will be included if the book is cloned.', 'pressbooks-book' ); ?>
+				<?php echo sprintf( '<button type="button" id="h5p-show-hide" class="btn btn-secondary btn-sm" show-all-text="%s" hide-all-text="%s" aria-label="%s"></button>', __( 'Expand all', 'pressbooks-book' ), __( 'Hide all', 'pressbooks-book' ), __( 'Expand all activities', 'pressbooks-book' ) ); ?>
 			</p>
-			<table class="table table-borderless" id="h5p-listing-table">
-				<thead>
-				<tr>
-					<th><?php echo __( 'ID', 'pressbooks-book' ); ?></th>
-					<th><?php echo __( 'Title', 'pressbooks-book' ); ?></th>
-					<th><?php echo __( 'Activity type', 'pressbooks-book' ); ?></th>
+		<?php endif; ?>
+		<?php if ( $filtered_does_not_exist ) : ?>
+			<p class="h5p-count">
+				<?php echo __( 'The H5P activity does not exist in this book.', 'pressbooks-book' ); ?>
```

Comment:
@SteelWagstaff do you consider this legend (below the title) is appropriate for invalid H5p_id parameters?

Reply:
I'd change to: 'The requested H5P activity does not exist in this book'